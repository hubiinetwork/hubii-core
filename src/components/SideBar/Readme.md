A variant of SideBar:

```js
const menuItems = [
  {
    to: '/wallets',
    icon: 'wallet',
    name: 'Wallet Manager'
  },
  {
    to: '/wallet',
    icon: 'wallet',
    name: 'Wallet detail'
  }
];
<SideBar menuItems={menuItems} logoSrc="../../../public/Images/corerz-logo.svg">
  <p>children</p>
</SideBar>;
```
