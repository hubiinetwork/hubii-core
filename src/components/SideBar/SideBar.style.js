import { Layout, Menu, Icon } from 'antd';
import styled from 'styled-components';

const SideBarLayout = styled(Layout)`
  .ant-layout-sider {
    background-color: ${({ theme }) => theme.palette.dark};
  }
  .ant-layout-sider.ant-layout-sider-collapsed {
    width: 72px !important;
    max-width: 72px !important;
    min-width: 72px !important;
  }
`;
const SideBarMenu = styled(Menu)`
  width: 72px;
  border-right: none;
  background-color: ${({ theme }) => theme.palette.dark};
  .menu-logo {
    min-height: 70px;
    margin: 0px !important;
    padding: 0px 15px !important;
    .logo-icon {
      width: 43px;
      margin-top: 15px !important;
    }
  }
  .menu-wallet {
    min-height: 70px;
    padding: 0px 20px !important;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  .menu-setting {
    bottom: 0px;
    width: 72px;
    height: 72px;
    line-height: 91px;
    padding: 0px 20px !important;
    margin-bottom: 0px !important;
    position: absolute !important;
    .setting-icon {
      font-size: 32px !important;
    }
  }
  .ant-menu-item a {
    opacity: 0.6;
    color: ${({ theme }) => theme.palette.dark2};
  }
  .ant-menu-item a:hover {
    opacity: 1;
  }
  .ant-menu-item-selected a {
    opacity: 1;
    color: ${({ theme }) => theme.palette.info} !important;
  }
  .ant-menu-item.menu-logo a {
    opacity: 1;
    color: ${({ theme }) => theme.palette.dark2};
  }
  .ant-menu-item-selected {
    opacity: 1;
    color: ${({ theme }) => theme.palette.info};
    background: ${({ theme }) => theme.palette.dark1} !important;
    border-right: 2px solid ${({ theme }) => theme.palette.info};
  }
  .ant-menu-item-selected.menu-logo {
    border-right: none;
    background: ${({ theme }) => theme.palette.dark} !important;
  }

  .ant-tooltip {
    display: none;
  }
`;

const MenuItemIcon = styled(Icon)`
  font-size: 32px !important;
  line-height: 70px !important;
`;
const Styledimg = styled.img`
  margin-top: 25px;
  margin-left: 5px;
`;
const StriimImage = styled.img`
  height: 30px;
  width: 35px;
`;
export { SideBarLayout, SideBarMenu, MenuItemIcon, Styledimg, StriimImage };
