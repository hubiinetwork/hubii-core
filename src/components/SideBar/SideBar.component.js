import { Icon, Layout, Menu } from 'antd';
import { Link, withRouter } from 'react-router-dom';
import React from 'react';
import PropTypes from 'prop-types';
import SvgIcon from '../ui/SvgIcon';
import darkTheme from '../../themes/darkTheme';

import NahmiiGrey from '../../../public/images/nahmii-token-grey.png';
import NahmiiGreen from '../../../public/images/nahmii-token-green.png';

import { SideBarLayout, SideBarMenu } from './SideBar.style';

const { Sider } = Layout;

/**
 * Side bar component to show features on a long, usually left, bar.
 */

export class SideBar extends React.Component {
  getroute() {
    const { location } = this.props;
    const arr = location.pathname.split('/');
    return `/${arr[1]}`;
  }

  render() {
    const { menuItems, logoSrc, children } = this.props;
    const { info, dark2 } = darkTheme.palette;
    const currentRoute = this.getroute();
    let selectedKeys;
    if (currentRoute.includes('wallet')) {
      selectedKeys = ['wallet'];
    } else if (currentRoute.includes('nahmii')) {
      selectedKeys = ['nahmii'];
    } else if (currentRoute.includes('dex')) {
      selectedKeys = ['dex'];
    } else if (currentRoute.includes('settings')) {
      selectedKeys = ['settings'];
    }
    return (
      <SideBarLayout>
        <Sider>
          <SideBarMenu
            defaultSelectedKeys={[menuItems[0].key]}
            mode="inline"
            onSelect={this.handleChange}
            selectedKeys={selectedKeys}
            forceSubMenuRender
          >
            <Menu.Item key="/" className="menu-logo">
              <Link to="/">
                <div>
                  <img
                    className="anticon anticon-wallet logo-icon"
                    src={logoSrc}
                    alt="logo"
                  />
                </div>
              </Link>
            </Menu.Item>
            {menuItems &&
              menuItems.map((menuItem) => (
                <Menu.Item key={menuItem.key} className="menu-wallet">
                  <Link to={menuItem.to}>
                    <div
                      style={{
                        display: 'flex',
                        justifyContent: 'center',
                        alignItems: 'center',
                      }}
                    >
                      {menuItem.icon === 'wallet' && (
                        <SvgIcon>
                          <path
                            d="M24,20 L24,21.3333333 C24,22.8 22.8,24 21.3333333,24 L2.66666667,24 C1.18666667,24 0,22.8 0,21.3333333 L0,2.66666667 C0,1.2 1.18666667,0 2.66666667,0 L21.3333333,0 C22.8,0 24,1.2 24,2.66666667 L24,4 L12,4 C10.52,4 9.33333333,5.2 9.33333333,6.66666667 L9.33333333,17.3333333 C9.33333333,18.8 10.52,20 12,20 L24,20 Z M12,17.3333333 L25.3333333,17.3333333 L25.3333333,6.66666667 L12,6.66666667 L12,17.3333333 Z M17.3333333,14 C16.2266667,14 15.3333333,13.1066667 15.3333333,12 C15.3333333,10.8933333 16.2266667,10 17.3333333,10 C18.44,10 19.3333333,10.8933333 19.3333333,12 C19.3333333,13.1066667 18.44,14 17.3333333,14 Z"
                            fill={
                              currentRoute === menuItem.to ||
                              currentRoute === '/wallet'
                                ? info
                                : dark2
                            }
                          />
                        </SvgIcon>
                      )}
                      {menuItem.icon === 'dex' && (
                      <SvgIcon viewBox="0 0 28 16">
                        <g
                          fill="none"
                          fillRule="evenodd"
                          transform="translate(-2 -8)"
                        >
                          <polygon points="0 0 32 0 32 32 0 32" />
                          <polygon
                            fill={currentRoute === menuItem.to ? info : dark2}
                            fillRule="nonzero"
                            points="21.333 8 24.387 11.053 17.88 17.56 12.547 12.227 2.667 22.12 4.547 24 12.547 16 17.88 21.333 26.28 12.947 29.333 16 29.333 8"
                          />
                        </g>
                      </SvgIcon>
                    )}
                      {menuItem.icon === 'nahmii-token' && (
                       currentRoute.includes('nahmii')
                       ?
                         <div>
                           <img alt="nahmii token" style={{ height: 'auto', width: '32px' }} src={NahmiiGreen} />
                         </div>
                      :
                         <div>
                           <img alt="nahmii token" style={{ height: 'auto', width: '32px' }} src={NahmiiGrey} />
                         </div>
                    )}
                    </div>
                  </Link>
                </Menu.Item>
                ))}
            <Menu.Item key="settings" className="menu-setting">
              <Link to="/settings">
                <div>
                  <Icon className="setting-icon" type="setting" />
                </div>
              </Link>
            </Menu.Item>
          </SideBarMenu>
        </Sider>
        {children}
      </SideBarLayout>
    );
  }
}
SideBar.propTypes = {
  /**
   * Array of menu items other than settings and logo
   */
  menuItems: PropTypes.arrayOf(
    PropTypes.shape({
      icon: PropTypes.string.isRequired,
      to: PropTypes.string.isRequired,
    })
  ),
  /**
   * Source of logo
   */
  logoSrc: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  location: PropTypes.object,
};

export default withRouter(SideBar);
