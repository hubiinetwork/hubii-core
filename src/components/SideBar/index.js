import SideBar from './SideBar.component';
export default SideBar;
