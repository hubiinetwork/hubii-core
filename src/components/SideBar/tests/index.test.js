import React from 'react';
import { shallow, mount } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import dark from '../../../themes/darkTheme';

import SideBar from '../index';

describe('<SideBar />', () => {
  const menuItems = [
    {
      to: '/wallets',
      icon: 'wallet',
      name: 'Wallet Manager'
    },
    {
      to: '/wallet',
      icon: 'striim',
      name: 'striim detail'
    },
    {
      to: '/dex',
      icon: 'dex',
      name: 'dex detail'
    }
  ];
  const props = {
    menuItems,
    logoSrc: '../../../public/images/hubii-core-logo.svg'
  };

  it('should render correctly', () => {
    const wrapper = shallow(
      <SideBar {...props}>
        <div />
      </SideBar>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should mount correctly', () => {
    const wrapper = mount(
      <Router>
        <ThemeProvider theme={dark}>
          <SideBar {...props}>
            <div />
          </SideBar>
        </ThemeProvider>
      </Router>
    );
    const sidebar = wrapper.find('.ant-layout-sider');
    expect(sidebar.props().className).toHaveLength(65);
  });
});
