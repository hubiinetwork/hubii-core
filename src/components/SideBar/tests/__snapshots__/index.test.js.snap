// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideBar /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withRouter(SideBar)
    logoSrc="../../../public/images/hubii-core-logo.svg"
    menuItems={
        Array [
            Object {
              "icon": "wallet",
              "name": "Wallet Manager",
              "to": "/wallets",
            },
            Object {
              "icon": "striim",
              "name": "striim detail",
              "to": "/wallet",
            },
            Object {
              "icon": "dex",
              "name": "dex detail",
              "to": "/dex",
            },
          ]
    }
>
    <div />
</withRouter(SideBar)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
